### ISSUE TITLE
POINT INDEX distance comparison with number higher than 11_936_226 fails "WGS84 invalid coordinate error"

### ISSUE THREAD

**Issue by ikwattro** [neo4j/neo4j#13384]
If I create a `POINT INDEX` and execute the following statement :

```cypher
WITH point({
    latitude: 17.235614712621427, 
    longitude: -170.95281462498815
}) AS centre
MATCH (n:EarthQuakeEvent)
WHERE point.distance(n.position, centre) < 11_936_227
RETURN count(*)
```

The following exception is thrown

```
Cannot create WGS84 point with invalid coordinate: [-180.0, -93.52370940305858]. Valid range for Y coordinate is [-90, 90].
```

It works for any distance calculation below `11_936_227`, so `11_936_226` and below work fine, don't ask me how I found the number 🙈

If the point index is not present, everything is fine as well.

---

**Neo4j Version:** 5.15
**Operating System:** Any
**API:** Any

### Steps to reproduce

- `CREATE DATABASE events WAIT`
- `CREATE POINT INDEX FOR (n:EarthQuakeEvent) ON n.position`
- Run the query above

### Expected behavior

The query should run successfully

### Actual behavior

```
Cannot create WGS84 point with invalid coordinate: [-180.0, -93.52370940305858]. Valid range for Y coordinate is [-90, 90].
```
is thrown

---

On a non empty database, I confirm that removing the index or using a range index returns my nodes correctly

Tested on 

- 5.15 ❌ 
- 5.13 ❌ 
- 5.10 ✅ 
-

**Comment by pontusmelke** [comment]
Hi @ikwattro, thanks for reporting. Will have a look.

**Comment by pontusmelke** [comment]
Hi, thanks for reporting this!
This bug has probably been hiding there since we introduced spatial indexes! It happens for distances so large that it covers both the north and the south pole,

![image](https://github.com/neo4j/neo4j/assets/88000/2e5fcf3c-ac65-4ca1-a180-12159f1a0332)

and we were sorely missing a test case for this 😞 

We have created a fix that should come out with upcoming releases.

Thanks again!
Best regards
Pontus
